src/sentry/utils/sentryappwebhookrequests.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import absolute_import
+
+import six
+import json
+
+from django.conf import settings
+from django.utils import timezone
+
+from sentry.utils import redis
+from sentry.models.sentryapp import VALID_EVENTS
+from sentry.models import Organization
+
+BUFFER_SIZE = 100
+
+
+class SentryAppWebhookRequestsBuffer(object):
+    """
+    Create a data structure to store basic information about Sentry App webhook requests in Redis
+    This should store the last 100 requests and last 100 errors (in different keys) for each event type, for each Sentry App
+    """
+
+    def __init__(self, sentry_app):
+        self.sentry_app = sentry_app
+
+        cluster_id = getattr(settings, "SENTRY_WEBHOOK_LOG_REDIS_CLUSTER", None)
+        if cluster_id is None:
+            self.client = redis.clusters.get("default").get_local_client(0)
+        else:
+            self.client = redis.redis_clusters.get(cluster_id)
+
+    def _convert_redis_request(self, request, event):
+        formatted_request = json.loads(request)
+        if "organization_id" in formatted_request:
+            org = Organization.objects.get(id=formatted_request["organization_id"])
+            formatted_request["organization"] = {"name": org.name, "slug": org.slug}
+            del formatted_request["organization_id"]
+
+        formatted_request["event_type"] = event
+        formatted_request["sentry_app_slug"] = self.sentry_app.slug
+
+        return formatted_request
+
+    def _add_to_buffer(self, buffer_key, item):
+        self.client.lpush(buffer_key, json.dumps(item))
+        self.client.ltrim(buffer_key, 0, BUFFER_SIZE - 1)
+
+    def add_request(self, response_code, org_id, event, url):
+        if event not in VALID_EVENTS:
+            return
+
+        request_key = self._get_redis_key(event)
+
+        time = timezone.now()
+        request_data = {
+            "date": six.binary_type(time),
+            "response_code": response_code,
```

Comment:
@nolachen what are we putting in `response_code` when the request times out and we don't get a response?

Reply:
Right now I’m putting those as 0 but that’s for all RequestExceptions which includes more than timeouts. Not sure if this is the best approach or if I should store a string actually explaining the error or something

Reply:
@nolachen I think that's acceptable since status code 0 is not a valid status code.